packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -893,6 +893,8 @@ export interface SdkPagingServiceMetadata<TServiceOperation extends SdkServiceOp
   continuationTokenResponseSegments?: (SdkServiceResponseHeader | SdkModelPropertyType)[];
   /** Segments to indicate how to get page items from response. */
   pageItemsSegments?: SdkModelPropertyType[];
+  /** Denotes which parameter is the page size parameter */
+  pageSizeParameter?: SdkMethodParameter;
```

Comment:
Since `@pageSize` could be defined nested, can we keep the same patten `pageSizeSegments` as other paging metadata?

Reply:
ooh that's a good point, thanks!